pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -2146,13 +2150,18 @@ func (cc *XController) Cleanup(ctx context.Context, appwrapper *arbv1.AppWrapper
 }
 func (cc *XController) getAppWrapper(namespace string, name string, caller string) (*arbv1.AppWrapper, error) {
 	klog.V(5).Infof("[getAppWrapper] getting a copy of '%s/%s' when called by '%s'.", namespace, name, caller)
-	apiCacheAWJob, err := cc.appWrapperLister.AppWrappers(namespace).Get(name)
-	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			klog.Errorf("[getAppWrapper] getting a copy of '%s/%s' failed, when called  by '%s', err=%v", namespace, name, caller, err)
+	var apiCacheAWJob *arbv1.AppWrapper
+	var err error
+	if cc.appwrapperInformer.Informer().HasSynced() {
+		apiCacheAWJob, err = cc.appWrapperLister.AppWrappers(namespace).Get(name)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				klog.Errorf("[getAppWrapper] getting a copy of '%s/%s' failed, when called  by '%s', err=%v", namespace, name, caller, err)
+			}
+			return nil, err
 		}
-		return nil, err
 	}
+
 	klog.V(5).Infof("[getAppWrapper] get a copy of '%s/%s' suceeded when called by '%s'", namespace, name, caller)
 	return apiCacheAWJob.DeepCopy(), nil
```

Comment:
If line 2155 evaluates to `false` (i.e., the informer has not synced) then `apiCacheAWJob` would be empty. Wouldn't calling a `DeepCopy` function on an empty object be a problem?

Reply:
Thanks, Created a new error object.

Reply:
Thanks